graveler/ref/branch_iterator_test.go

```diff
@@ -0,0 +1,97 @@
+package ref_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/go-test/deep"
+	"github.com/treeverse/lakefs/graveler"
+	"github.com/treeverse/lakefs/graveler/ref"
+	"github.com/treeverse/lakefs/testutil"
+)
+
+func TestPGBranchIterator(t *testing.T) {
+	r, db := testRefManagerWithDB(t)
+	branches := []graveler.BranchID{"a", "aa", "b", "c", "e", "d"}
+	ctx := context.Background()
+	testutil.Must(t, r.CreateRepository(ctx, "repo1", graveler.Repository{
+		StorageNamespace: "s3://foo",
+		CreationDate:     time.Now(),
+		DefaultBranchID:  "master",
+	}, graveler.Branch{}))
+
+	// prepare data
+	for _, b := range branches {
+		testutil.Must(t, r.SetBranch(ctx, "repo1", b, graveler.Branch{CommitID: "c1"}))
+	}
+
+	t.Run("listing all branches", func(t *testing.T) {
+		iter := ref.NewBranchIterator(ctx, db, "repo1", 3)
```

Comment:
3 divides 6, so this doesn't test partial batching.  Can you add another branch on l.16, or change "3" to "4" or "5", please?

Reply:
keeping branch iterator working on 7 items.